code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -10335,10 +10335,39 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 		if(!locate(/obj/effect/decal/cleanable/purpledrank) in T)
 			new /obj/effect/decal/cleanable/purpledrank(T)
 
+/datum/reagent/punctualite
+	name = "Punctualite"
+	id = PUNCTUALITE
+	description = "Nicknamed mad chemist's alarm clock. Explodes on the hour when being actively metabolized. "
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#8d8791" //rgb: 200, 165, 220
+	custom_metabolism = 0 //Wouldn't be much fun if it all got metabolized beforehand
+	var/currentHour = 0 //The hour it was introduced into the system so it doesn't blow right away
+
+/datum/reagent/punctualite/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+	if(!currentHour)
+		currentHour = floor(world.time / 36000) + 1
+	if(prob(5) && prob(5)) //0.25% chance per tick
+		M.visible_message("<span class='notice'>Tick.. tock...</span>")
+	if((floor(world.time / 36000) + 1) > currentHour)
```

Comment:
world.timeofday will actually just give you the formatted time
time2text(world.timeofday, "hh") will pull out just the hour in question

much less messy than this math

Reply:
Oh thank goodness, I absolutely hated how I wrote this. Thanks.